lms/djangoapps/email_marketing/signals.py

```diff
@@ -54,18 +58,15 @@ def add_email_marketing_cookies(sender, response=None, user=None,
         if sailthru_content:
             post_parms['cookies'] = {'anonymous_interest': sailthru_content}
 
-    try:
-        sailthru_client = SailthruClient(email_config.sailthru_key, email_config.sailthru_secret)
-        log.info(
-            'Sending to Sailthru the user interest cookie [%s] for user [%s]',
-            post_parms.get('cookies', ''),
-            user.email
-        )
-        time_before_call = datetime.datetime.now()
+    time_before_call = datetime.datetime.now()
+    sailthru_response = get_email_cookies_via_sailthru.delay(post_parms)
 
-        sailthru_response = \
-            sailthru_client.api_post("user", post_parms)
+    try:
+        sailthru_response.get(timeout=SAILTHRU_TIMEOUT, propagate=True)
+        sailthru_response = sailthru_response.result
 
+    except TimeoutError as exc:
+        log.error("Timeout error while attempting to obtain cookie from Sailthru: %s", unicode(exc))
     except SailthruClientError as exc:
```

Comment:
I think my last comment was a mistake about SailthruClientError.
I think we should catch this exception in the Celery task itself, instead of trying to catch it outside of the task.
I don't believe we are able to catch this exception outside of the celery task